src/main/java/com/marklogic/client/ext/schemasloader/impl/QbvDocumentFileProcessor.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2023 MarkLogic Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.marklogic.client.ext.schemasloader.impl;
+
+import com.marklogic.client.DatabaseClient;
+import com.marklogic.client.document.XMLDocumentManager;
+import com.marklogic.client.eval.ServerEvaluationCall;
+import com.marklogic.client.ext.file.DocumentFile;
+import com.marklogic.client.ext.file.DocumentFileProcessor;
+import com.marklogic.client.ext.helper.LoggingObject;
+import com.marklogic.client.io.DOMHandle;
+import com.marklogic.client.io.Format;
+import com.marklogic.client.io.StringHandle;
+import org.jdom2.Element;
+import org.jdom2.input.SAXBuilder;
+import org.jdom2.output.DOMOutputter;
+import org.springframework.util.FileCopyUtils;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.util.ArrayList;
+import java.util.List;
+
+public class QbvDocumentFileProcessor extends LoggingObject implements DocumentFileProcessor {
+
+	public static final String QBV_COLLECTION = "http://marklogic.com/xdmp/qbv";
```

Comment:
This constant can stay public, but make all the others private. And make all the fields and methods private that don't need to be public (this is all part of trying to limit the public API, which I made too broad in a lot of places).